calcom/cal.com
# fix: Avatar slug and cal links for cross org users

## What does this PR do?

- Fix wrong avatar and booking link for non-org member in an org team(Scenario-1)
	- When migrating a user to an org we end up in this scenario where team is migrated to Org but some of it's users aren't
		- It's important to do it because we should infer the avatar and link from the user and not from the booking domain he is on.This completely avoids the issues where we see different user's avatar for a user.
- Fix wrong avatar shown on Team Availability


## Type of change





## How should this be tested?
- [x] Scenario-1: Non org member in an org team
	- 	Create an org(O1) and a team(T1) inside it.
	-  Add a member to the team T1
	- Now go to db-studio and set organizationId to null for the user and also remove membership of the user with the Org. It creates a scenario where non-org member is member of a team.
	- Now, go to org booking page(o1.cal.local:3000) and then go to the team page in the org. You will still see that member there. It should correctly show it's avatar, and when you click on it, it should correctly take you to it's booking link which would be on app.cal.local:3000 and not o1.cal.local:3000
- [x] Scenario-2: Two users with same username(one in org and other outside org)
	- Login to org O1 and invite a new user teampro@o1.com which would have username teampro(same as seeded user)
	- Go to availability and switch to Team Availability, notice that the avatar for teampro is not of Org user `teampro` but of non-org user `teampro`. This bug is fixed here.


## Mandatory Tasks


## Checklist
- I haven't added tests that prove my fix is effective or that my feature works
